beeline/test_trace.py

```diff
@@ -316,6 +317,59 @@ def test_add_trace_field_propagates(self):
                 'app.more': 'data!',
         })
 
+    def test_add_rollup_field_propagates(self):
+        m_client = Mock()
+        tracer = SynchronousTracer(m_client)
+        tracer._run_hooks_and_send = Mock()
+
+        span1 = tracer.start_trace(context={'name': 'root'})
+        event1 = m_client.new_event.return_value
+
+        span2 = tracer.start_span(context={'name': 'middle'})
+        event2 = m_client.new_event.return_value
+
+        span3 = tracer.start_span(context={'name': 'inner1'})
+        event3 = m_client.new_event.return_value
+
+        tracer.add_rollup_field('database_ms', 17)
+        tracer.add_rollup_field('calories', 180)
+        tracer.add_rollup_field('database_ms', 23.1)
+
+        event3.add_field.reset_mock()
+        tracer.finish_span(span3)
+        event3.add_field.assert_has_calls([
+            call('database_ms', 17.0 + 23.1),
+            call('calories', 180.0),
+            call('duration_ms', ANY),
+        ])
+
+        span4 = tracer.start_span(context={'name': 'inner2'})
+        event4 = m_client.new_event.return_value
+
+        tracer.add_rollup_field('calories', 120)
+
+        event4.add_field.reset_mock()
+        tracer.finish_span(span4)
+        event4.add_field.assert_has_calls([
+            call('calories', 120.0),
+            call('duration_ms', ANY),
+        ])
+
+        event2.add_field.reset_mock()
+        tracer.finish_span(span2)
+        event2.add_field.assert_has_calls([
+            # This intermediate span doesn't get any rollup fields.
+            call('duration_ms', ANY),
+        ])
+
+        event1.add_field.reset_mock()
+        tracer.finish_span(span1)
+        event1.add_field.assert_has_calls([
```

Comment:
I think we may need to set `any_order=True` in these calls to `assert_has_calls`, since the ordering of python dicts is not deterministic in 3.5. For what we're checking here, I think ordering is not important so not enforcing it is probably OK.

Reply:
Good catch, done. Tests pass now.